packages/neon_framework/test/user_status_bloc_test.dart

```diff
@@ -0,0 +1,218 @@
+        json.encode(
+          {
+            'ocs': {
+              'meta': {'status': '', 'statuscode': 0},
+              'data': <dynamic>[],
+            },
+          },
+        ),
+        200,
+      );
+
+  final requests = <String, Map<String, Response Function(Map<String, String> queryParameters)>>{
+    '/ocs/v2.php/apps/user_status/api/v1/predefined_statuses': {
+      'get': (_) => predefinedStatusesResponse(),
+    },
+    '/ocs/v2.php/apps/user_status/api/v1/user_status/message/predefined': {
+      'put': (queryParameters) {
+        message = queryParameters['messageId'];
+        messageIsPredefined = true;
+        clearAt = int.parse(queryParameters['clearAt']!);
+        return statusResponse();
+      },
+    },
+    '/ocs/v2.php/apps/user_status/api/v1/user_status/message/custom': {
+      'put': (queryParameters) {
+        message = queryParameters['message'];
+        messageIsPredefined = false;
+        icon = queryParameters['statusIcon'];
+        clearAt = int.parse(queryParameters['clearAt']!);
+        return statusResponse();
+      },
+    },
+    '/ocs/v2.php/apps/user_status/api/v1/user_status/status': {
+      'put': (queryParameters) {
+        status = queryParameters['statusType']!;
+        statusIsUserDefined = true;
+        return statusResponse();
+      },
+    },
+    '/ocs/v2.php/apps/user_status/api/v1/user_status/message': {
+      'delete': (_) {
+        messageIsPredefined = false;
+        message = null;
+        icon = null;
+        clearAt = null;
+        return emptyResponse();
+      },
+    },
+    '/ocs/v2.php/apps/user_status/api/v1/heartbeat': {
+      'put': (queryParameters) {
+        status = queryParameters['status']!;
+        statusIsUserDefined = false;
+        return statusResponse();
+      },
+    },
+  };
+
+  return Account(
+    serverURL: Uri.parse('https://example.com'),
+    username: 'test',
+    password: 'test',
+    httpClient: MockClient((request) async {
+      final call = requests[request.url.path]?[request.method];
+      if (call != null) {
+        return call(request.url.queryParameters);
+      }
+
+      throw Exception(request);
+    }),
+  );
+}
+
+void main() {
+  late Account account;
+  late UserStatusBloc bloc;
+
+  setUpAll(() {
+    final platform = MockNeonPlatform();
+    when(() => platform.canUseWindowManager).thenReturn(false);
+    NeonPlatform.mocked(platform);
+  });
+
+  setUp(() {
+    account = mockUserStatusAccount();
+    bloc = UserStatusBloc(account);
+  });
+
+  test('refresh', () async {
+    expect(
+      bloc.predefinedStatuses.transformResult((e) => e.single.id),
+      emitsInOrder([
+        Result<String>.loading(),
+        Result.success('id'),
+        Result.success('id').asLoading(),
+        Result.success('id'),
+      ]),
+    );
+    // The delay is necessary to avoid a race condition with loading twice at the same time
+    await Future<void>.delayed(const Duration(milliseconds: 1));
```

Comment:
@Leptopoda I saw you simply used expectLater without awaiting it in https://github.com/nextcloud/neon/pull/1500, I guess I need to do the same thing here

Reply:
Nope wrong, the delay is really necessary to avoid the race condition.